index.d.ts

```diff
@@ -0,0 +1,39 @@
+interface Options {
+	/**
+	 * Applies a function to the JSON string before parsing.
+	 */
+	beforeParse?: Function
+	/**
+	 * Prescribes how the value originally produced by parsing is transformed, before being returned.
+	 * See the [JSON.parse docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse#Using_the_reviver_parameter) for more.
+	 */
+	reviver?: Function
+}
+
+/**
+ * Read and parse a JSON file.
+ *
+ * Strips UTF-8 BOM, uses graceful-fs, and throws more helpful JSON errors.
+ *
+ * @example
+ * import * as loadJsonFile from 'load-json-file';
+ *
+ * loadJsonFile.sync('foo.json')
+ * console.log(json);
+ */
+export function sync(filepath: string, options?: Options): void;
+
+/**
+ * Read and parse a JSON file.
+ *
+ * Strips UTF-8 BOM, uses graceful-fs, and throws more helpful JSON errors.
+ *
+ * @example
+ * import loadJsonFile from 'load-json-file';
+ *
+ * (async () => {
+ * 	await loadJsonFile('foo.json');
+ * 	console.log('done');
+ * })();
+ */
+export default function loadJsonFile(filepath: string, options?: Options): Promise<void>;
```

Comment:
It returns a `Promise<any>`. What we can do however is make it generic like this

```ts
export default function loadJsonFile<T = any>(filepath: string, options?: Options): Promise<T>;
```

Then the user can supply the type that it returns.

```ts
interface Foo {
    value: string;
}

const data = await loadJsonFile<Foo>('data.json');
//=> data is typed as `Foo`
```

Reply:
👍 you're correct, it doesn't return `Promise<void>` 🤦‍♂️ 

I like using the generic approach 👍